### DISCUSSION TITLE
Issue with Shape Alignment in JointJS due to Reference Path

### DISCUSSION THREAD

**Discussion by VADDIJASWANT** [clientIO/joint#2380] [Q&A] [answered]
### Introduction

I am in the process of migrating a project that uses SVG for rendering shapes to a new one that utilizes JointJS. In our previous product, we rendered a shape directly using SVG with the path M0 35L70 35. However, with JointJS, we are unable to provide a path directly to the shapes and are instead passing this path as a reference to a JointJS shape.

During this migration, I’ve observed that the shapes from our old product are not aligning correctly in the new one. It appears that JointJS is interpreting the reference path as 0 0 to 70 0, leading to alignment issues.

I’m seeking guidance on how to rectify this problem. Is there a way to adjust the reference path or another workaround to ensure correct alignment of shapes?

For a more direct comparison, I you can refer this [CodePen ](https://codepen.io/VADDIJASWANT/pen/ExGNLvO?editors=1010)that renders both the SVG and JointJS elements.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
It is not clear from the example what the desired behavior should look like.

*Hint:* don't use the `refD` attribute (it expands the path to fit into the element bounding box), try using the `d` expression with `calc()` expressions instead. This way you will be able to define a line that does not start in the top left corner of the element.
